crates/anvil/src/eth/api.rs

```diff
@@ -1076,6 +1076,10 @@ impl EthApi {
     /// Waits for a transaction to be included in a block and returns its receipt (no timeout).
     async fn await_transaction_inclusion(&self, hash: TxHash) -> Result<ReceiptResponse> {
         let mut stream = self.new_block_notifications();
+        // Check if the transaction is already included before listening for new blocks.
+        if let Some(receipt) = self.backend.transaction_receipt(hash).await? {
```

Comment:
does it matter if it's before or after `self.new_block_notifications()`?

Reply:
then we still have the same race condition only a smaller window